polyfills/fm/client/fmradio.js

```diff
@@ -0,0 +1,553 @@
+        }
+      }
+    };
+  };
+
+  var enabled, antennaAvailable, frequencyLowerBound, frequencyUpperBound,
+    channelWidth, frequency;
+
+  window.navigator.mozFMRadio = {
+    fakeEnabled: false,
+
+    fakeAntennaAvailable: false,
+
+    fakeFrequencyLowerBound: 87.5,
+
+    fakeFrequencyUpperBound: 108,
+
+    fakeChannelWidth: 0.1,
+
+    disable: function fm_disable() {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'disable'
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    },
+
+    enable: function fm_enable(frequency) {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'enable',
+        params: frequency
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    },
+
+    seekUp: function fm_seekUp() {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'seekUp'
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    },
+
+    seekDown: function fm_seekDown() {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'seekDown'
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    },
+
+    cancelSeek: function fm_cancelSeek() {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'cancelSeek'
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    },
+
+    setFrequency: function fm_setFrequency(freq) {
+      var request = new FakeDOMRequest();
+      var data = {
+        operation: 'invoke',
+        method: 'setFrequency',
+        params: freq
+      };
+
+      request.serialize = serializeDOMRequest.bind(request, data);
+      navConnPromise.then(conn => conn.sendObject(request));
+      return request;
+    }
+  };
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'enabled', {
+    get: function() {
+      // Enabled value should be updated
+      if (typeof enabled !== 'undefined' && enabled !== null) {
+        return enabled;
+      }
+
+      var data = {
+        operation: 'get',
+        name: 'enabled'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      return this.fakeEnabled;
+    },
+
+    set: function(newValue) {
```

Comment:
Hmm those attributes are readonly... they should not have setters! Just store the real value on another attribute (if this exists because of line 91)